src/VisualStudio/Core/Def/Implementation/Watson/WatsonReporter.cs

```diff
@@ -119,39 +126,48 @@ public static void ReportNonFatal(Exception exception)
             session.PostEvent(faultEvent);
         }
 
-        private static List<string> CaptureLogFiles()
+        private static List<string> CollectServiceHubLogFilePaths()
         {
-            var result = new List<string>();
-            CollectServiceHubLogFiles(result);
-            CaptureFilesInMemory(result);
-            return result;
-        }
+            var paths = new List<string>();
 
-        private static void CollectServiceHubLogFiles(List<string> paths)
-        {
             try
             {
                 var logPath = Path.Combine(Path.GetTempPath(), "servicehub", "logs");
                 if (!Directory.Exists(logPath))
                 {
-                    return;
+                    return paths;
                 }
 
                 // attach all log files that are modified less than 1 day before.
                 var now = DateTime.UtcNow;
                 var oneDay = TimeSpan.FromDays(1);
 
-                foreach (var file in Directory.EnumerateFiles(logPath, "*.log"))
+                foreach (var path in Directory.EnumerateFiles(logPath, "*.log"))
                 {
                     try
                     {
-                        var lastWrite = File.GetLastWriteTimeUtc(file);
+                        var name = Path.GetFileNameWithoutExtension(path);
```

Comment:
Do we really need to trim the name if you're just going to do name.Contains() anyways?

Reply:
The path is a full path and might contain arbitrary strings.